realm/src/androidTest/java/io/realm/RealmAsyncQueryTests.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+import android.os.Looper;
+import android.test.InstrumentationTestCase;
+
+import junit.framework.AssertionFailedError;
+
+import java.util.Date;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import io.realm.entities.AllTypes;
+import io.realm.entities.NonLatinFieldNames;
+
+public class RealmAsyncQueryTests extends InstrumentationTestCase {
+
+    //TODO use Junit4 & define a rule to setup the threading, to avoid
```

Comment:
We cannot use JUnit4 AFAIK as that require us mocking all Android framework classes, especially the Looper :(